src/net_processing.h

```diff
@@ -42,6 +42,16 @@ class PeerManager : public CValidationInterface, public NetEventsInterface
                                              CTxMemPool& pool, bool ignore_incoming_txs);
     virtual ~PeerManager() { }
 
+    /**
+     * Attempt to manually fetch block from a given peer. We must already have the header.
+     *
+     * @param[in]  id       The peer id
+     * @param[in]  hash     The block hash
+     * @param[in]  pindex   The blockindex
```

Comment:
why are you passing the hash twice? the blockindex already holds its own hash, no?

Also, `pindex` is not a pointer, so should be called `block_index`.

Reply:
That may have gotten mixed up over the course of rebases and changes, I'll take a second look.